packages/@aws-cdk/aws-stepfunctions-tasks/lib/lambda/run-lambda-task.ts

```diff
@@ -12,7 +12,7 @@ export interface RunLambdaTaskProps {
    *
    * @default - No payload
    */
-  readonly payload?: { [key: string]: any };
+  readonly payload?: sfn.TaskInput;
```

Comment:
Is it worth introducing a new field for this? (Let's say, `input?: sfn.TaskInput`) and `@deprecate`ing the old one?

Are you okay taking the breakage?

-----

Next question: since we're touching this anyway, does it make sense to change the `@default` while we're at it?

Reply:
I'm ok with the breaking change, given the migration is straight forward.

Why do you suggest changing the default? It seems to me that the current default makes sense, no?

Reply:
I don't see any problems with no payload being the default as long as there is a code example that shows how to set `"Payload.$": "$"` as @rix0rrr suggests.

`$` (the entire input) could be a sensible default value, mostly out of convenience for users of the "old-style" `InvokeFunction` which always sent the effective input as the payload. I noted some caveats in https://github.com/aws/aws-cdk/issues/7371#issuecomment-615935009.

It may be worth adding test coverage for setting payload to `$` since that's such a common use case.

Reply:
I'm also okay with taking the breakage here for the same reason as @nija-at 

good point: I do think the `@default` should be the entire task input while we're at it as I think that's a more common usage than sending the empty object into a Lambda. I've left it as is for now, but happy to make that change

Reply:
I think we should optimize for the most common use case. What do we expect MOST people to do? @wong-a I would hope your team has some info on the most common ways people use each task type.

Furthermore, naively I would say that changing payload to `$` by default is a non-breaking change. If people leave it empty, today they would get no parameters, so not expect them and not read them, so adding more parameters should be fine.

There's also the case of what the default `outputPath` should be. Feels like for maximum convenience it should be `$.Payload`, and similar for all other task types. But changing that is not backwards compatible so I guess that ship has sailed, huh...?

Reply:
Hey @wong-a, since we got you here anyway... what is the official term for the JSON blob that travels along with the execution, which is updated using inputPath, outputPath, etc?

I was struggling to find a good, clear and concise name for it while writing the library, so ultimately settled on something unsatisfying like `Data`. I wonder what you call it internally?

I guess we could do ExecutionState or something, but any term I can come up with feels so generic as to be worthless... :/

Reply:
> What do we expect MOST people to do? @wong-a I would hope your team has some info on the most common ways people use each task type.

The old-style Lambda is the most used Task type, which uses the entire effective input and I think most customers would be happy with it as the default. If someone wants to pass an empty object as the payload they can easily do so by passing an empty object as the value of `Payload`.

> what is the official term for the JSON blob that travels along with the execution, which is updated using inputPath, outputPath, etc?

There's no special name for it. It's generally referred to within the [ASL spec](https://states-language.net/spec.html#filters) and Step Functions documentation as "state input" or output.

Reply:
thanks for that valuabe input @wong-a 

I flipped this to default to `$` as I agree with @rix0rrr that changing the payload from empty object to the state input is not *likely* to break the lambda being invoked as it wasn't expecting to read any input.

I will call it out in the breaking change entry.